test/functional/ui/messages_spec.lua

```diff
@@ -3138,3 +3138,210 @@ it('pager works in headless mode with UI attached', function()
     -- More --^                              |
   ]])
 end)
+
+describe('progress-message', function()
```

Comment:
add a `it('validates')` block that tests all/most of the errors https://github.com/neovim/neovim/blob/58060c2340a52377a0e1d2b782ce1deef13b2b9b/test/functional/lua/vim_spec.lua#L2921-L2925